firstvoices/backend/tests/test_apis/base_api_test.py

```diff
@@ -44,3 +50,60 @@ def get_detail_endpoint(self, site_slug, key):
 
     def setup_method(self):
         self.client = APIClient()
+
+    @pytest.mark.django_db
+    def test_list_404_site_not_found(self):
+        user = factories.get_non_member_user()
+        self.client.force_authenticate(user=user)
+
+        response = self.client.get(self.get_list_endpoint(site_slug="missing-site"))
+
+        assert response.status_code == 404
+
+    @pytest.mark.django_db
+    def test_list_403_site_not_visible(self):
+        user = factories.get_non_member_user()
+        self.client.force_authenticate(user=user)
+        site = factories.SiteFactory.create(visibility=Visibility.TEAM)
+
+        response = self.client.get(self.get_list_endpoint(site_slug=site.slug))
+
+        assert response.status_code == 403
+
+    @pytest.mark.django_db
+    def test_list_empty(self):
+        user = factories.get_non_member_user()
+        self.client.force_authenticate(user=user)
+        site = factories.SiteFactory.create(visibility=Visibility.PUBLIC)
+
+        response = self.client.get(self.get_list_endpoint(site_slug=site.slug))
+
+        assert response.status_code == 200
+        response_data = json.loads(response.content)
+        assert len(response_data["results"]) == 0
+        assert response_data["count"] == 0
+
+    @pytest.mark.django_db
+    def test_list_team_access(self):
+        site = factories.SiteFactory.create(visibility=Visibility.TEAM)
+        user = factories.get_non_member_user()
+        factories.MembershipFactory.create(user=user, site=site, role=Role.ASSISTANT)
+        self.client.force_authenticate(user=user)
+
+        response = self.client.get(self.get_list_endpoint(site.slug))
+
+        assert response.status_code == 200
+        response_data = json.loads(response.content)
+        assert response_data["results"] == []
+
+    @pytest.mark.django_db
+    def test_detail_404_unknown_key(self):
+        site = factories.SiteFactory.create(visibility=Visibility.PUBLIC)
+        user = factories.get_non_member_user()
+        self.client.force_authenticate(user=user)
+
+        response = self.client.get(
+            self.get_detail_endpoint(site_slug=site.slug, key="fake-key")
+        )
+
+        assert response.status_code == 404
```

Comment:
You found even more reusable tests! Nice!

Reply:
+1 to this. Thanks for adding these!